packages/@cdklabs/typewriter/src/renderer/typescript.ts

```diff
@@ -411,11 +411,10 @@ export class TypeScriptRenderer extends Renderer {
     throw new Error(`Symbol ${sym} (in ${sym.scope}) not visible from ${this.scopes[0]} (missing import?)`);
   }
 
-  protected renderFunction(func: CallableDeclaration) {
+  protected renderFunction(func: CallableDeclaration & TypeDeclaration) {
```

Comment:
I don't think this is correct, we want to accept all `CallableDeclarations` not, just the ones that are also a `TypeDeclaration`.

Reply:
I think we keep the signature as it is.

Then w can have a guard `isExported(x): is { exported: true } { ... }` to check if  `exported` exists and is true.